assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidget.js

```diff
@@ -67,15 +70,35 @@ function DashboardTopEarningPagesWidget( { Widget, WidgetReportZero, WidgetRepor
 			limit: 5,
 		};
 
+		const adSenseLinked = select( MODULES_ANALYTICS ).getAdsenseLinked();
+
 		return {
-			isAdSenseLinked: select( MODULES_ANALYTICS ).getAdsenseLinked(),
+			isAdSenseLinked: adSenseLinked,
 			analyticsMainURL: select( MODULES_ANALYTICS ).getServiceReportURL( 'content-publisher-overview', generateDateRangeArgs( { startDate, endDate } ) ),
 			data: select( MODULES_ANALYTICS ).getReport( args ),
 			error: select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [ args ] ),
 			loading: ! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [ args ] ),
+			isAdblockerActive: adSenseLinked && select( MODULES_ADSENSE ).isAdblockerActive(),
 		};
 	} );
 
+	if ( isAdblockerActive ) {
+		return (
+			<Widget
+				Footer={ () => (
+					<SourceLink
+						className="googlesitekit-data-block__source"
+						name={ _x( 'Analytics', 'Service name', 'google-site-kit' ) }
+						href={ analyticsMainURL }
+						external
+					/>
```

Comment:
Let's extract it into an anonymous component and use it here and in the happy path return to avoid duplicate code:

```javascript
const sourceLink = () => (
    <SourceLink ... />
);

if ( isAdblockerActive ) {
    return (
        <Widget Footer={ sourceLink }>
            ...
        </Widget>
    );
}

...

return (
    <Widget
        noPadding
        Footer={ sourceLink }
    >
        ...
     </Widget>
);
```